contributing/testing-best-practices.md

```diff
@@ -32,6 +32,15 @@
 - Data sources are tested in the same tests as the resources, e.g. [commonChecks](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/66c44e62c9afe04ffe8be0dbccaec682bab830e6/internal/service/searchindex/resource_search_index_test.go#L262-L263).
 - Helper functions such as `resource.TestCheckTypeSetElemNestedAttrs` or `resource.TestCheckTypeSetElemAttr` can be used to check resource and data source attributes more easily, e.g. [resource_serverless_instance_test.go](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/66c44e62c9afe04ffe8be0dbccaec682bab830e6/internal/service/serverlessinstance/resource_serverless_instance_test.go#L61).
 
+### Cloud Gov tests
+
+1. Use [`PreCheck: PreCheckGovBasic`](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-250271_cloud_gov/internal/testutil/acc/pre_check.go#L98)
+2. Use the [`acc.ConfigGovProvider`](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-250271_cloud_gov/internal/testutil/acc/provider.go#L61) together with your normal terraform config
+3. Modify the `checkExist` and `CheckDestroy` to use `acc.ConnV2UsingGov`
+4. Follow naming convention:
+   1. `TestAccGovProject_withProjectOwner`, note prefix: `TestAccGov`
```

Comment:
```suggestion
   1. `TestAccProject_govWithProjectOwner`, note prefix: `TestAccGov`
```
not a strong opinion but like the idea of leaving the first part before the underscore only with the resource, and second part giving detail of the test case (as shown in [tf docs](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/testcase#creating-acceptance-tests-functions))

Reply:
I think using it in the first part makes `regex` selection simpler. 
Although with this approach it might not be necessary 🤔 WDYT?

Reply:
> I think using it in the first part makes regex selection simpler

Not sure if I understand this point given we wont need to identify gov tests from others

Reply:
Good discussion. Added a bit more context in c888aff

Reply:
I think we should be executing tests by resource and not by environment (gov vs. non-gov). For example, if we change something in the project resource, we would want to run tests in gov too to ensure nothing broke.

Would `Test(Acc|Mig)Gov` do that with the current implementation?